### PR TITLE
Let freetype handle antialiasing

### PR DESCRIPTION
**Pull request by Scyak**
Antialiasing is no longer done manually, instead the font is rendered antialiased via freetype when using bpp=8. This improves positioning of the antialiased character.

### CODE REVIEW

File: ttf2ugui.c [hunk 1]
```diff
@@ -283,19 +283,11 @@ static UG_FONT *convertFont(const char *font, int dpi, float fontSize,int bitsPe
...
-  {
-    case 1: bpp_mul = 1; break;
-    case 8: bpp_mul = 16; break;
-    default:
-    {
-       fprintf(stderr, "Bits per pixel must be 1 or 8, not %d!!\n", bitsPerPixel);
-       exit(1);
-    }break;
-
+  if(bitsPerPixel != 1 && bitsPerPixel != 8) {
```
Comment (AriZuu) [inline ttf2ugui.c]: formatting: space between if & parenthesis.

File: ttf2ugui.c [hunk 2]
```diff
@@ -325,16 +317,16 @@ static UG_FONT *convertFont(const char *font, int dpi, float fontSize,int bitsPe
...
   else
-    error = FT_Set_Pixel_Sizes(face, 0, fontSize * bpp_mul);
+    error = FT_Set_Pixel_Sizes(face, 0, fontSize);
   if (error) {
 
     fprintf(stderr, "set pixel sizes err %d\n", error);
     exit(1);
   }
 
-  int i, j,i_idx,j_idx;
```
Comment (AriZuu) [inline ttf2ugui.c:337]: formatting: space after comma

File: ttf2ugui.c [hunk 3]
```diff
@@ -348,12 +340,19 @@ static UG_FONT *convertFont(const char *font, int dpi, float fontSize,int bitsPe
  * First found out how big character bitmap is needed. Every character
  * must fit into it so that we can obtain correct character positioning.
  */
+
+    int flags = FT_LOAD_RENDER;
+
+    if(bitsPerPixel == 1) {
```
Comment (AriZuu) [inline ttf2ugui.c]: formatting: space after if

File: ttf2ugui.c [hunk 4]
```diff
@@ -412,38 +410,29 @@ static UG_FONT *convertFont(const char *font, int dpi, float fontSize,int bitsPe
...
-            }
-            for(j_idx = 0; j_idx < bpp_mul;j_idx++) {
-                if (j * bpp_mul + j_idx >= face->glyph->bitmap.width) {
-                  break;
-                }
+        uint8_t *bits = (uint8_t *) face->glyph->bitmap.buffer;
 
-                uint8_t *bits = (uint8_t *) face->glyph->bitmap.buffer;
-                uint8_t b = bits[(i*bpp_mul+i_idx) * face->glyph->bitmap.pitch + ((j*bpp_mul+j_idx) / 8)];
+        if(bitsPerPixel == 8) {
```
Comment (AriZuu) [inline ttf2ugui.c]: formatting: space after if

**Review by AriZuu — COMMENTED** [review]
Some notes about code formatting, would be great if you'd polish those spots.

**Comment by Scyak** [comment]
I ran clang-format across the whole thing to get everything consistent, making some assumptions about the way it was supposed to be configured based on what was already there. But if you'd prefer to leave the formatting of the stuff I didn't change untouched I can also remove that commit again.

File: ttf2ugui.c (hunk 2 above)
Comment (Scyak) [reply ttf2ugui.c:337]: Those parts of the line were removed anyways

**Comment by AriZuu** [comment]
clang-format is fine. Merged. Thanks again.

**Pull request merged**